File: server/models/Attendance.js
const mongoose = require('mongoose');

const attendanceSchema = new mongoose.Schema({
  studentId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'Student',
    required: true
  },
  course: {
    type: String,
    required: true,
    trim: true
  },
  batch: {
    type: String,
    required: true,
    trim: true
  },
  date: {
    type: Date,
    required: true
  },
  status: {
    type: String,
    enum: ['Present', 'Absent'],
    default: 'Present'
  }
}, {
  timestamps: true
});

module.exports = mongoose.model('Attendance', attendanceSchema);

File: mobile/lib/core/models/attendance.dart
class Attendance {
  final String id;
  final DateTime date;
  final String status; // Present, Absent, Late
  final String course;

  Attendance({
    required this.id,
    required this.date,
    required this.status,
    required this.course,
  });

  factory Attendance.fromJson(Map<String, dynamic> json) => Attendance(
        id: json['_id'] ?? json['id'] ?? '',
        date: DateTime.parse(json['date'] ?? DateTime.now().toIso8601String()),
        status: json['status'] ?? 'Absent',
        course: json['course'] ?? '',
      );

  Map<String, dynamic> toJson() => {
        'id': id,
        'date': date.toIso8601String(),
        'status': status,
        'course': course,
      };
}
